virtcontainers/clh.go

```diff
@@ -877,461 +767,256 @@ func MaxClhVCPUs() uint32 {
+	ctx, cancel := context.WithTimeout(context.Background(), clhApiTimeout*time.Second)
+	defer cancel()
 
-	cmdline.args = append(cmdline.args, cscNet)
+	cl := c.client()
 
-	networks := ""
-	netIndex := 1
-	for _, net := range o.networks {
-		cnet := "tap=tap" + strconv.FormatUint(uint64(netIndex), 10)
-		if net.ip != "" && net.mask != "" {
-			cnet += ",ip=" + net.ip + ",mask=" + net.mask
-		}
-		if net.mac != "" {
-			cnet += ",mac=" + net.mac
-		}
-
-		if net.iommu {
-			cnet += ",iommu=on"
-		}
-		if netIndex > 1 {
-			networks += "," + cnet
-		} else {
-			networks += cnet
+	if c.config.Debug {
+		bodyBuf, err := json.Marshal(c.vmconfig)
+		if err != nil {
+			return err
 		}
-		netIndex++
+		c.Logger().WithField("body", string(bodyBuf)).Warn("VM config")
 	}
-	cmdline.args = append(cmdline.args, networks)
-}
+	_, err := cl.CreateVM(ctx, c.vmconfig)
 
-//****************************************
-// The log file
-//****************************************
-type CLILogFile struct {
-	path string
-}
+	if err != nil {
+		return openApiClientError(err)
+	}
 
-func (o *CLILogFile) Build(cmdline *CommandLine) {
+	info, _, err := cl.VmInfoGet(ctx)
 
-	if o.path != "" {
-		cmdline.args = append(cmdline.args, cscLogFile)
-		cmdline.args = append(cmdline.args, o.path)
+	if err != nil {
+		return openApiClientError(err)
 	}
-}
-
-//****************************************
-// The kernel command line
-//****************************************
-type CLICmdline struct {
-	params []Param
-}
 
-func (o *CLICmdline) Build(cmdline *CommandLine) {
+	c.Logger().Debugf("VM state after create: %#v", info)
 
-	cmdline.args = append(cmdline.args, cscCmdline)
+	if info.State != "Created" {
+		return fmt.Errorf("VM state is not running after create")
+	}
 
-	var paramBuilder strings.Builder
-	for _, p := range o.params {
-		paramBuilder.WriteString(p.Key)
-		if len(p.Value) > 0 {
+	c.Logger().Debug("Booting VM")
+	_, err = cl.BootVM(ctx)
 
-			paramBuilder.WriteString("=")
-			paramBuilder.WriteString(p.Value)
-		}
-		paramBuilder.WriteString(" ")
+	if err != nil {
+		return openApiClientError(err)
 	}
-	cmdline.args = append(cmdline.args, strings.TrimSpace(paramBuilder.String()))
 
-}
+	info, _, err = cl.VmInfoGet(ctx)
 
-//**********************************
-// The Default Builder
-//**********************************
-type DefaultCLIBuilder struct {
-	console   *CLIConsole
-	serial    *CLISerialConsole
-	apiSocket *CLIAPISocket
-	cpus      *CLICpus
-	memory    *CLIMemory
-	kernel    *CLIKernel
-	disk      *CLIDisk
-	fs        *CLIFs
-	rng       *CLIRng
-	logFile   *CLILogFile
-	vsock     *CLIVsock
-	cmdline   *CLICmdline
-	nets      *CLINets
-}
+	if err != nil {
+		return openApiClientError(err)
+	}
 
-func (d *DefaultCLIBuilder) AddKernelParameters(params []Param) {
+	c.Logger().Debugf("VM state after boot: %#v", info)
 
-	if d.cmdline == nil {
-		d.cmdline = &CLICmdline{}
+	if info.State != "Running" {
+		return fmt.Errorf("VM state is not created after create")
```

Comment:
This doesn't look correct based on the clause which this comment is within? Can you clarify here, and line 908?

Reply:
Sorry for the confusion. Those are typos in the error message, while the API calls and checks are correct.